File: library/src/main/java/com/pengrad/telegrambot/model/request/InputLocationMessageContent.java
package com.pengrad.telegrambot.model.request;

import java.io.Serializable;

/**
 * Stas Parshin
 * 06 May 2016
 */
public class InputLocationMessageContent extends InputMessageContent implements Serializable {
    private final static long serialVersionUID = 0L;

    private Float latitude;
    private Float longitude;
    private Float horizontal_accuracy;
    private Integer live_period;
    private Integer heading;
    private Integer proximity_alert_radius;

    public InputLocationMessageContent(Float latitude, Float longitude) {
        this.latitude = latitude;
        this.longitude = longitude;
    }

    public InputLocationMessageContent horizontalAccuracy(float horizontalAccuracy) {
        horizontal_accuracy = horizontalAccuracy;
        return this;
    }

    public InputLocationMessageContent livePeriod(Integer livePeriod) {
        live_period = livePeriod;
        return this;
    }

    public InputLocationMessageContent heading(int heading) {
        this.heading = heading;
        return this;
    }

    public InputLocationMessageContent proximityAlertRadius(int proximityAlertRadius) {
        proximity_alert_radius = proximityAlertRadius;
        return this;
    }
}

File: core/src/commonMain/kotlin/com/pengrad/telegrambot/model/request/InputLocationMessageContent.kt
package com.pengrad.telegrambot.model.request

import java.io.Serializable

/**
 * Stas Parshin
 * 06 May 2016
 */
class InputLocationMessageContent(private val latitude: Float?, private val longitude: Float?) : InputMessageContent(), Serializable {
    private var live_period: Int? = null

    fun livePeriod(livePeriod: Int?): InputLocationMessageContent {
        live_period = livePeriod
        return this
    }

    companion object {
        private const val serialVersionUID = 0L
    }
}
